### DISCUSSION TITLE
New features suggestions

### DISCUSSION THREAD

**Discussion by lchauvin** [Kitware/VolView#601] [Ideas]
Hello,

I'm planning to develop the following features, but I don't know if it's something you'd be interested to integrate them in the codebase:

- Synchronize images and cameras: this is useful when looking at different modalities of a same subject (images registered), and you want to compare them. For example, zoom in one part of the brain in t1w image, and quickly switch to the same location for t2w. By keeping cameras in sync, it's easy to do. I intend to have an icon in top of the images in the image browser to toggle sync for cameras and images 
- Left / Right arrows to cycle through images. Up / down arrows to change slices
- A button on top of the toolbar to collapse the image menu (to focus on images)

I already implemented some of these features. Let me know if it's something you'd be interested to integrate in the code base.

Thank you.

**Comment by floryst** [comment]
Those sound like good features to add! We'd be happy to review and integrate your changes.

* camera synchronization: will there be a constraint on which images are synchronizable? Or will we merely designate a base image as the camera source? (This is related to the existing layers functionality, but still distinct.)
* It would also be good to evaluate whether we want left/right buttons somewhere on the UI for image cycling.

**Reply by lchauvin** [reply]
My idea would be to allow image synchronization only if all images are in the same space (I think there is already a function to check that), and sync all images then, at least in a first version. Maybe if need be, we could only sync images that are selected, but it's gonna get complex to show which images are synchronized with which.

**Reply by floryst** [reply]
Sounds good to me!
